backend/Squirrel.ConsoleApp/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.SignalR.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Squirrel.ConsoleApp.BL.Interfaces;
+using Squirrel.ConsoleApp.Models;
+
+namespace Squirrel.Core.WebAPI.Extensions;
+
+public static class ApplicationBuilderExtensions
+{
+    private static HubConnection? _hubConnection;
+    private static string? _clientId;
+
+    public static void InitializeFileSettings(this IApplicationBuilder app)
+    {
+        var fileService = app.ApplicationServices.GetRequiredService<IConnectionFileService>();
+        fileService?.CreateEmptyFile();
+    }
+
+    public static void RegisterHubs(this IApplicationBuilder app, IConfiguration config)
+    {
+        var fileService = app.ApplicationServices.GetRequiredService<IClientIdFileService>();
+        var getActionsService = app.ApplicationServices.GetRequiredService<IGetActionsService>();
+
+        _clientId = fileService?.GetClientId();
+
+        _hubConnection = new HubConnectionBuilder()
+            .WithUrl(config.GetSection("SignalRSettings")["HubConnectionUrl"])
+            .WithAutomaticReconnect()
+            .Build();
+
+        // Use once at OnConnectedAsync hub event
+        _hubConnection.On<string>("SetClientId", (guid) =>
+        {
+            if (!string.IsNullOrEmpty(_clientId))
+            {
+                return;
+            }
+            fileService?.SetClientId(guid);
+            _clientId = guid;
+        });
+
+        // DELETE: Just for testing
+        _hubConnection.On("ExecuteQuery", (string clientId) =>
+        {
+            if (string.IsNullOrEmpty(_clientId))
+            {
+                return;
+            }
+
+            if (!clientId.Equals(_clientId))
+            {
+                return;
+            }
```

Comment:
you can replace it with a sinle `if` statement or extract a method

Reply:
Leave this implementation